packages/gui/src/components/nfts/NFTPreview.tsx

```diff
@@ -58,24 +314,35 @@ export default function NFTPreview(props: NFTPreviewProps) {
     fit = 'cover',
     background: Background = Fragment,
     hideStatusBar = false,
+    isPreview = false,
+    isCompact = false,
+    metadata,
+    isOffer = false,
   } = props;
 
-  const [loaded, setLoaded] = useState(false);
   const hasFile = dataUris?.length > 0;
   const file = dataUris?.[0];
+  let extension: string = new URL(file).pathname.split('.').slice(-1)[0];
+  if (!extension.match(/^[a-zA-Z0-9]+$/)) {
+    extension = '';
+  }
+
+  const [loaded, setLoaded] = useState(false);
+  const { isValid, isLoading, error, thumbnail } = useNFTHash(
+    nft,
+    isPreview,
+    metadata,
+  );
+
   const [ignoreError, setIgnoreError] = usePersistState<boolean>(
     false,
     `nft-preview-ignore-error-${nft.$nftId}-${file}`,
   );
-  const [ignoreSizeLimit, setIgnoreSizeLimit] = usePersistState<boolean>(
-    false,
-    `nft-preview-ignore-size-limit-${nft.$nftId}-${file}`,
-  );
-  const { isValid, isLoading, error } = useNFTHash(nft, ignoreSizeLimit);
```

Comment:
This was added fairly recently as a bug fix (see PR #899). The bug was that a large image wasn't showing the "Image Hash Mismatch" status label.

Testing out the branch, I'm seeing the buggy behavior prior to the bug fix, so this is a regression.